codersdk/client.go

```diff
@@ -119,33 +120,54 @@ func readBodyAsError(res *http.Response) error {
 		helper = "Try logging in using 'coder login <url>'."
 	}
 
-	if strings.HasPrefix(contentType, "text/plain") {
-		resp, err := io.ReadAll(res.Body)
-		if err != nil {
-			return xerrors.Errorf("read body: %w", err)
+	resp, err := io.ReadAll(res.Body)
+	if err != nil {
+		return xerrors.Errorf("read body: %w", err)
+	}
```

Comment:
I think it'd be nicer to only `io.ReadAll` if the mime isn't json, that way we don't need to buffer the entire body for json responses as well.

Reply:
This is needed for returning invalid JSON to the user unfortunately